src/test/threadpool_tests.cpp

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <util/string.h>
+#include <util/threadpool.h>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_AUTO_TEST_SUITE(threadpool_tests)
+
+constexpr auto TIMEOUT_SECS = std::chrono::seconds(120);
+
+template <typename T>
+void WaitFor(const std::vector<std::future<T>>& futures, const std::string& context)
+{
+    for (size_t i = 0; i < futures.size(); ++i) {
+        if (futures[i].wait_for(TIMEOUT_SECS) != std::future_status::ready) {
+            throw std::runtime_error("Timeout waiting for: " + context + ", task index " + util::ToString(i));
+        }
+    }
+}
```

Comment:
As mentioned [before](https://github.com/bitcoin/bitcoin/pull/26966#discussion_r2424865671), I think these could be simplified to `std::span`, `std::string_view` and `strprintf`, something like:
```suggestion
template <typename T>
void WaitFor(std::span<const std::future<T>> futures, std::string_view context)
{
    for (size_t i{0}; i < futures.size(); ++i) {
        if (futures[i].wait_for(TIMEOUT) != std::future_status::ready) {
            throw std::runtime_error(strprintf("Timeout waiting for: %s, task index %d", context, i));
        }
    }
}
```

----

But if we extract each test to a separate unit (as suggested below), we won't need the context parameter anymore.
The thread index also doesn't really help in case of the failure, which would enable us to simplify this to:
```suggestion
template <typename T>
void WaitFor(std::span<const std::future<T>> futures)
{
    for (const auto& f : futures) {
        BOOST_REQUIRE(f.wait_for(TIMEOUT) == std::future_status::ready);
    }
}
```